get_data_meteorological_multiworker.py

```diff
@@ -0,0 +1,249 @@
+import os
+import sys
+import time
+import traceback
+import json
+import random
+import csv
+import queue
+import logging
+from multiprocessing import Pool
+import multiprocessing
+
+import requests
+
+
+TOKEN_FILE = "tokens.listing"
+cdo_tokens = multiprocessing.Queue()
+STATION_LOOKUP={}
+
+
+logger = logging.getLogger("noaa_climate_pull")
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+handlerF = logging.FileHandler("noaa_climate_pull.log")
+handlerF.setFormatter(formatter)
+handlerF.setLevel(logging.DEBUG)
+
+handlerC = logging.StreamHandler()
+handlerC.setFormatter(formatter)
+handlerC.setLevel(logging.INFO)
+
+# Add the handlers to the logger
+logger.addHandler(handlerF)
+logger.addHandler(handlerC)
+
+
+def get_timeseries_batched(entity_desc, datasetid, stn_id, year_start, year_end, request_uri, out_file_json, out_file_csv, field_list, extra_params=None):
+
+    result_count = 1000000 # does not matter atm, offset of 0 will always dominate
+    current_offset = 0
+    result_limit = 1000
+    no_results = False
+
+    list_obtained = []
+
+    license_key = cdo_tokens.get(block=True)
+    logger.debug(f"Using License key {license_key}")
+
+    stn_name = STATION_LOOKUP[stn_id]["stn_name"]
+    logger.info(f"Pulling for station {stn_id}, readings from {year_start}-{year_end}, location: {stn_name}")
+
+    try:
+
+        for year_current in range(year_start, year_end+1):
+            no_results = False
+            current_offset = 0
+
+            head = {'token': license_key}
+
+            while not no_results and (current_offset < result_count):
```

Comment:
I think for maintainability, we should make a separate function to make a call to the API

```
def parse_api_response(year, **kwargs):
    response = requests.get(url, params=params, **kwargs)
    < additional error checking > 
    return response.json()

def def get_timeseries_batched(...):
    results = []
    for year_current in range(year_start, year_end+1):
        response = parse_api_response(year)
        results.extend(response['results'])
```